### DISCUSSION TITLE
Table 树形数据更新 Items 后未能刷新显示

### DISCUSSION THREAD

**Discussion by pankey888** [dotnetcore/BootstrapBlazor#3168] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

在【Table 树形数据展示】最后一个Demo中，添加一个按钮，用于更新TreeItems，代码如下：
```
private async Task OnTableButtonClick()
{
    TreeItems = TreeFoo.GenerateFoos(LocalizerFoo, 3);
    // 在Id 为 2的节点插入1条 子项
    TreeItems.AddRange(TreeFoo.GenerateFoos(LocalizerFoo, 1, 2, 100));
    StateHasChanged();
    await Task.CompletedTask;
}
```
当我点击该按钮，发现表格中的数据并未变化。但当修改表格参数 IsTree="false"后，点击按钮，表格中会显示变化的数据。

### Expected Behavior

能支持在树形结构下动态更新Items参数后正确显示新的数据。

**Comment by bb-auto** [comment]
@pankey888 Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

**Comment by pankey888** [comment]
可通过本示例代码进行复现：
[BootstrapBlazorApp5.zip](https://github.com/dotnetcore/BootstrapBlazor/files/14796654/BootstrapBlazorApp5.zip)

**Comment by ArgoZhang** [comment, marked as answer]
@pankey888 Table 组件功能比较多，为了性能 `IsTree="true"` 的时候内部使用了缓存技术，你这样更改是无法更改 UI 显示的，需要增加一个参数 `key`

```
<Table @key="TreeItems" TItem="TreeFoo" IsBordered="true" ShowToolbar="true" IsStriped="true" TreeIcon="fa-solid fa-circle-chevron-right" TreeExpandIcon="fa-solid fa-circle-chevron-right fa-rotate-90"
       Items="@TreeItems" IsTree="true" TreeNodeConverter="@TreeNodeConverter" OnTreeExpand="@OnTreeExpand">
    <TableColumns>
        <TableColumn @bind-Field="@context.Name" Width="360" />
        <TableColumn @bind-Field="@context.DateTime" Width="180" />
        <TableColumn @bind-Field="@context.Address" />
        <TableColumn @bind-Field="@context.Count" />
    </TableColumns>
    <TableToolbarTemplate>
        <Button Color="Color.Primary" IsOutline="false" Text="修改数据" OnClick="@OnTableButtonClick"></Button>
    </TableToolbarTemplate>
</Table>
```

请点击下面 `make as answer`

**Comment by ArgoZhang** [comment]
讨论区不需要关闭，方便别人查阅
